### PR TITLE
Minor improvements

### PR DESCRIPTION
**Pull request by Rombur**
These are just some minors improvements. The first commit requires C++14. Given that gcc 6, which will  reach end of life later this month, defaults to C++14, this change shouldn't be too controversial. The second commit adds a few pictures to CONTRIBUTING.md to better explain the work flow (https://github.com/Rombur/dealii/blob/dank/CONTRIBUTING.md).

### CODE REVIEW

**Comment by drwells** [comment]
OK if the tester passes 👍

File: cmake/checks/check_01_cxx_features.cmake [hunk 1]
```diff
@@ -154,6 +154,11 @@ IF(NOT DEFINED DEAL_II_WITH_CXX14 OR DEAL_II_WITH_CXX14)
 ENDIF()
 
 
+IF (NOT DEAL_II_HAVE_CXX14)
+  MESSAGE(FATAL_ERROR "Deal.II requires C++14.")
+ENDIF()
```
Comment (bangerth) [inline cmake/checks/check_01_cxx_features.cmake]: Is this your Easter Egg/April Fool's joke?

Comment (Rombur) [inline cmake/checks/check_01_cxx_features.cmake]: :-)

**Comment by bangerth** [comment]
Nice try. I can't figure out whether @drwells just played along or didn't read that far in the patch. I thought about all of those pictures that I may not have put them in there myself, but I could go along with it if others who are younger than me wanted to ;-)

**Comment by drwells** [comment]
By the time I had read the patch the tester had already failed. I was tempted to make a remark about requiring C++17 but I figured that might give the joke away  :->

**Pull request closed without merging**